src/services/implementation/ClientServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package services.implementation;
+
+import entities.Client;
+import enums.StatusEnum;
+import exceptions.services.EntityNotFoundException;
+import exceptions.services.IdNullPointerException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import services.interfaces.IClientInterface;
+
+public class ClientServiceImpl implements IClientInterface {
+  private List<Client> allClients;
+  private static ClientServiceImpl instance;
+  private Long id;
+
+  private ClientServiceImpl() {
+    this.allClients = new ArrayList<>();
+    this.id = 0L;
+  }
+
+  public static ClientServiceImpl getInstance() {
+    return Optional.ofNullable(instance).orElseGet(ClientServiceImpl::new);
+  }
+
+  @Override
+  public Optional<Client> insert(Client client) {
+    client.setId(this.id);
+    client.setCode("C" + this.allClients.size());
```

Comment:
Create a utility function to build the code, use `String.format`

Reply:
solved